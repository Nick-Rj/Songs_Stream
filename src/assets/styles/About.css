.About {
  height: inherit;
  background-color: rgba(29, 63, 214, 0.507);
  opacity: 0.9;
  width: 100vw;
}

.About h1,
h2,
h4 {
  color: rgb(56, 219, 105);
}

.About h3,
p {
  color: black;
}

.About h1 {
  margin-top: 2rem;
}

.About #info {
  color: black !important;
}
