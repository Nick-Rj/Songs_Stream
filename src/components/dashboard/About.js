import React, { useState } from "react";
import {
  FloatingLabel,
  Container,
  Form,
  Row,
  Button,
  Col,
  Modal,
  Card,
} from "react-bootstrap";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import styled, { keyframes } from "styled-components";
import {
  pulse,
  tada,
  lightSpeedIn,
  rollIn,
  bounce,
  bounceIn,
  rubberBand,
  fadeIn,
  flash,
  zoomInUp,
  slideInUp,
  jello,
  swing,
} from "react-animations";
import "../../assets/styles/About.css";

const AnimateAbout = styled.div`
  animation: 1.5s ${keyframes`${bounce}`};
`;

const About = ({ userState }) => {
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const username = userState.isLoggedIn ? userState.user.firstName : "User";
  return (
    <div className="About">
      <Container id="header">
        <Row>
          <Col>
            <Link to="/songsLibrary">
              <Button variant="warning" id="backToLib">
                Back to Songs
              </Button>
            </Link>
          </Col>
          <Col xs={2} lg={4}></Col>
          <Col xs={2} lg={4}></Col>
        </Row>
      </Container>
      <AnimateAbout>
        <Container>
          <Row>
            <Col>
              <h1>Hi {username}!</h1>
              <h2>About our Music Platform!</h2>
              <h4>
                This is a lightweight Music App where you can add your favourite
                songs and listen on the go!
              </h4>
              <h4>
                Add your favourite songs, delete songs, update your songs, add
                them into a playlist and a lot more...
              </h4>
              <br />
              <br />
              <h2>Hop in and experience this new app like never before!</h2>
            </Col>
          </Row>
          <Row>
            <Col>
              <Button variant="success" onClick={handleShow}>
                Know About Developer!
              </Button>
            </Col>
            <Modal
              show={show}
              onHide={handleClose}
              size="lg"
              aria-labelledby="contained-modal-title-vcenter"
              centered
            >
              <Modal.Header closeButton>
                <Modal.Title id="contained-modal-title-vcenter">
                  Developer's Info
                </Modal.Title>
              </Modal.Header>
              <Modal.Body>
                <div id="info">
                  <Card style={{ width: "100%" }}>
                    <Card.Body>
                      <Card.Title>Nikhil Raj</Card.Title>
                      <Card.Subtitle className="mb-2 text-muted">
                        MERN Stack Developer
                      </Card.Subtitle>
                      <Card.Text>
                        <h4 style={{ color: "black" }}>Skills:</h4>
                        <p>
                          HTML, CSS, JavaScript, ES6, React.js, Redux, TDD,
                          Node.js, Express.js, MongoDB, Git{" "}
                        </p>
                      </Card.Text>
                      <Card.Link
                        href="https://github.com/Nick-Rj"
                        target="_blank"
                      >
                        GitHub
                      </Card.Link>
                      <Card.Link
                        href="https://www.linkedin.com/in/nikhil-raj-289b0a171"
                        target="_blank"
                      >
                        LinkedIn
                      </Card.Link>
                    </Card.Body>
                  </Card>
                </div>
              </Modal.Body>
              <Modal.Footer>
                <Button onClick={handleClose}>Close</Button>
              </Modal.Footer>
            </Modal>
          </Row>
        </Container>
      </AnimateAbout>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    userState: state.user,
  };
};

export default connect(mapStateToProps, {})(About);
