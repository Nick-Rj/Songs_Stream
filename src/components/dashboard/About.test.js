import { shallow } from "enzyme";
import { expect, it } from "@jest/globals";

import About from "./About";

import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";

const mockStore = configureMockStore();
const store = mockStore({
  userState: {
    loading: false,
    users: [],
    user: {},
    error: "",
    isLoggedIn: false,
  },
  songData: { loading: false, songs: [], error: "" },
  playlistData: { loading: false, playlists: [], error: "" },
});

it("rendering full DOM for About Component", () => {
  const wrapper = shallow(
    <Provider store={store}>
      <About />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
