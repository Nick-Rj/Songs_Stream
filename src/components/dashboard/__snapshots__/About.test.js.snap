// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering full DOM for About Component 1`] = `ShallowWrapper {}`;
